MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -5,67 +5,144 @@ import MapboxMobileEvents
-            modifiedEventDictionary["originalEstimatedDistance"] = round(session.originalRoute.distance)
-            modifiedEventDictionary["originalEstimatedDuration"] = round(session.originalRoute.expectedTravelTime)
+            originalGeometry = Polyline(coordinates: geometry).encodedPolyline
+            originalEstimatedDistance = round(session.originalRoute.distance)
+            originalEstimatedDuration = round(session.originalRoute.expectedTravelTime)
         }
         if let geometry = session.currentRoute.coordinates {
-            modifiedEventDictionary["geometry"] = Polyline(coordinates: geometry).encodedPolyline
-            modifiedEventDictionary["estimatedDistance"] = round(session.currentRoute.distance)
-            modifiedEventDictionary["estimatedDuration"] = round(session.currentRoute.expectedTravelTime)
+            self.geometry = Polyline(coordinates: geometry).encodedPolyline
+            estimatedDistance = round(session.currentRoute.distance)
+            estimatedDuration = round(session.currentRoute.expectedTravelTime)
         }
         
-        modifiedEventDictionary["distanceCompleted"] = round(session.totalDistanceCompleted + routeProgress.distanceTraveled)
-        modifiedEventDictionary["distanceRemaining"] = round(routeProgress.distanceRemaining)
-        modifiedEventDictionary["durationRemaining"] = round(routeProgress.durationRemaining)
+        distanceCompleted = round(session.totalDistanceCompleted + routeController.routeProgress.distanceTraveled)
+        distanceRemaining = round(routeController.routeProgress.distanceRemaining)
+        durationRemaining = round(routeController.routeProgress.durationRemaining)
         
-        modifiedEventDictionary["rerouteCount"] = session.numberOfReroutes
-
-        modifiedEventDictionary["volumeLevel"] = Int(AVAudioSession.sharedInstance().outputVolume * 100)
-        modifiedEventDictionary["screenBrightness"] = Int(UIScreen.main.brightness * 100)
-
-        modifiedEventDictionary["batteryPluggedIn"] = UIDevice.current.batteryState == .charging || UIDevice.current.batteryState == .full
-        modifiedEventDictionary["batteryLevel"] = UIDevice.current.batteryLevel >= 0 ? UIDevice.current.batteryLevel * 100 : -1
-        modifiedEventDictionary["applicationState"] = UIApplication.shared.applicationState.telemetryString
+        rerouteCount = session.numberOfReroutes
+        
+        volumeLevel = Int(AVAudioSession.sharedInstance().outputVolume * 100)
+        screenBrightness = Int(UIScreen.main.brightness * 100)
+        
+        batteryPluggedIn = UIDevice.current.batteryState == .charging || UIDevice.current.batteryState == .full
+        batteryLevel = UIDevice.current.batteryLevel >= 0 ? UIDevice.current.batteryLevel * 100 : -1
+        applicationState = UIApplication.shared.applicationState.telemetryString
+    }
+    
+    func convertToDictionary() -> [String: Any] {
```

Comment:
Nit: The verb implies a mutating function. `convertedToDictionary` or `asDictionary` would be more idiomatic.